core/src/main/scala/kafka/log/Log.scala

```diff
@@ -292,6 +305,20 @@ class Log(val dir: File,
    */
   def numberOfSegments: Int = segments.size
 
+
+  /**
+   * Returns compiled information about segments
+   */
+  def segmentsInfo: util.ArrayList[String] = {
+    val list = new util.ArrayList[String]()
+    logSegments.foreach(
+      seg => list.add("baseOffset=%s, created=%s, logSize=%s, indexSize=%s".format(
+        seg.baseOffset, seg.created, seg.size, seg.index.sizeInBytes()
+      ))
+    )
+    list
```

Comment:
I agree, but JConsole doesn't really show attribute values for Scala collections decorated as lists. Haven't digged deeper, but I suspect in this case it needs Scala classes in the classpath in order to show attributes correctly. It seems to me that the benefit of being able to use JConsole out of the box outweighs simplicity of the code in this case.

Example:
```scala
import com.yammer.metrics.Metrics
import com.yammer.metrics.core.Gauge
import com.yammer.metrics.reporting.JmxReporter
import scala.collection.JavaConverters._

object Test
{
  def main(args: Array[String]) {
    val gaugeA = new Gauge[java.util.List[String]] {
      override def value(): java.util.List[String] = {
        val x = new java.util.LinkedList[String]()
        x.add("a")
        x.add("b")
        x.add("c")
        x
      }
    }

    val gaugeB = new Gauge[java.util.List[String]] {
      override def value(): java.util.List[String] = {
        val a = List("a", "b", "c")
        a.asJava
      }
    }

    Metrics.defaultRegistry().newGauge(this.getClass, "testA", gaugeA)
    Metrics.defaultRegistry().newGauge(this.getClass, "testB", gaugeB)

    val reporter = new JmxReporter(Metrics.defaultRegistry())
    reporter.start()

    Thread.sleep(1000 * 3600)
  }
}
```

![testa](https://cloud.githubusercontent.com/assets/1021271/14337318/97d58ac6-fc21-11e5-93f4-ed340ca9213b.png)
![testb](https://cloud.githubusercontent.com/assets/1021271/14337322/9bfcbed0-fc21-11e5-941b-5443705da9a3.png)